packages/@aws-cdk/aws-rds/lib/database-secret.ts

```diff
@@ -55,5 +66,17 @@ export class DatabaseSecret extends secretsmanager.Secret {
         excludeCharacters: props.excludeCharacters ?? DEFAULT_PASSWORD_EXCLUDE_CHARS,
       },
     });
+
+    if (props.overrideLogicalId) {
+      const hash = crypto.createHash('md5');
+      hash.update(JSON.stringify({
+        path: this.node.path, // make it unique
+        // Use here the options that influence the password generation
+        excludeCharacters: props.excludeCharacters,
```

Comment:
We don't need to hash props that do not influence a password regeneration. If those change we can keep the same secret.

Reply:
So shouldn't we include `username` and `masterSecret`? https://github.com/aws/aws-cdk/blob/122c68c89838a9f5c4755d909635d8407f9a8019/packages/%40aws-cdk/aws-rds/lib/database-secret.ts#L48-L56

Reply:
I actually thought about that:

`masterSecret` is only used for the multi user rotation scheme so a secret with it never represents master credentials and we can just le CF update the secret.

`username` I'm less sure but as RDS doesn't support changing the username without replacing the database, do we need to cover this? Not hashing it sorts of protect the database from a username change?

Reply:
> `masterSecret` is only used for the multi user rotation scheme so a secret with it never represents master credentials and we can just let CF update the secret. There's no reason to force a replacement. Also this prop cannot be set when using the `Credentials` class.

Not sure I understand the "is only used for the multi user rotation scheme"... what about this?

```ts
    const secret = new rds.DatabaseSecret(stack, 'UserSecret', {
      username: 'user-username',
      masterSecret,
      excludeCharacters: '"@/\\',
    });

    const masterSecret = new rds.DatabaseInstance(stack, 'MasterSecret', {
      // ...
      credentials: rds.Credentials.fromSecret(secret),
    });
```

?

Reply:
Yes you can actually write that code. But (1) if the user creates its own secret we are never going to control its logical id with a hash and so never control when its replaced instead of updated and (2) specifying a master secret is used to create username (vs master) credentials that use a master secret for their rotation in the multi rotation scheme. It doesn't make sense to specify a master secret for master credentials.

Reply:
OK. If you're confident that we only need `excludeCharacters`, I'm fine with that 🙂.

Reply:
If at the some point we allow to customize something like `passwordLength` then it should be added.

Reply:
Can you include that as a comment on the code?